store/tikv/gcworker/gc_worker.go

```diff
@@ -511,6 +512,73 @@ func resolveLocks(ctx goctx.Context, store tikv.Storage, safePoint uint64, ident
 	return nil
 }
 
+type gcTask struct {
+	startKey []byte
+	endKey []byte
+	safePoint uint64
+}
+
+type gcResult struct {
+	successRegions int
+	failedRegions int
+	lastErr error
+}
+
+func genNextGCTask(store tikv.Storage, bo *tikv.Backoffer, safePoint uint64, key kv.Key) (*gcTask, error) {
+	loc, err := store.GetRegionCache().LocateKey(bo, key)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	task := &gcTask {
+		startKey: key,
+		endKey: loc.EndKey,
+		safePoint: safePoint,
+	}
+	return task, nil
+}
+
+func gcTaskWorker(store tikv.Storage, taskCh chan *gcTask, resultCh chan *gcResult) {
```

Comment:
How about defining a struct type named `gcTaskWorker`, then add a method `run`?

Reply:
Yes, I think it is a good idea.